repository/user/user_repository_impl.go

```diff
@@ -44,6 +45,18 @@ func (repository *userRepositoryImpl) Login(ctx context.Context, user user_entit
 	return loggedInUser, nil
 }
 
+func (repository *userRepositoryImpl) Edit(ctx context.Context, user user_entity.User) error {
+	if err := repository.IsExist(ctx, user.Id, user.Role); err != nil {
+		return err
+	}
+	query := "UPDATE users SET nip = $1, name = $2, identity_card_scan_img = $3 WHERE id = $4 RETURNING id"
```

Comment:
mas di code" yg execute IsExist itu bisa ga ya dijadikan 1 query? tujuannya biar ga call db 2x gitu. kyak gini contohnya

```sql
UPDATE users SET nip = $1, name = $2, identity_card_scan_img = $3 WHERE id = $4
AND EXISTS (SELECT 1 FROM users WHERE role = $5 AND id = $6 AND is_deleted = false)
RETURNING id
```

untuk $5 dan $6 itu buat `user.Role` dan `user.Id`

Reply:
Nah ini kenapa dibedakan soalnya Beda error antara role nya beda sama userIdnya not found

Reply:
401 sama 404 ya mas? ini error role-nya beda itu terkait role orang yg mengedit?

Reply:
OH nggak. Kalau yang di edit itu role IT kan return nya 400? Bukan 404

Reply:
Eh ternyata udah berubah 404 semua ya.

Reply:
iya mas, 404 semua udahan. jadi harusnya bisa kan ya cuma 1 query gitu. kalo role bukannya jg sudah d handle di middleware yg di router?